crates/ruff_linter/src/rules/pylint/rules/class_variable_slots_conflict.rs

```diff
@@ -0,0 +1,207 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{
+    Expr, ExprDict, ExprList, ExprName, ExprSet, ExprStarred, ExprStringLiteral, ExprTuple, Stmt,
+    StmtAssign, StmtClassDef, StmtFunctionDef,
+};
+use ruff_text_size::TextRange;
+
+use crate::checkers::ast::Checker;
+
+use rustc_hash::FxHashMap;
+
+/// ### What it does
+///
+///
+/// ### Why is this bad?
+///
+///
```

Comment:
Would you mind completing the *Why is this bad* section. It would help me understand the purpose of the rule and triaging it.